### PR TITLE
gem DataFormats for digi errors

### PR DESCRIPTION
**Pull request by jshlee**
splitting PR #21259 
 - adding ME0 to FEDNumbering
 - gem GEB and VFat status digis

@slava77 @kpedro88 @hyunyong

### CODE REVIEW

**Comment by civanch** [comment]
please test

File: DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    uint32_t getZeroSup() const {return ZeroSup_;}   ///<Returns Zero Suppression flags
+    uint8_t  getInputID() const {return InputID_;}   ///<Returns GLIB input ID
+    uint16_t getVwh() const     {return Vwh_;}       ///<Returns VFAT word count (size of VFAT payload)
+    uint16_t getErrorC() const  {return ErrorC_;}    ///<Returns thirteen flags in GEM Chamber Header
+
+    uint16_t getOHCRC() const   {return OHCRC_;}     ///<Returns OH CRC 
+    uint16_t getVwt() const     {return Vwt_;}       ///<Returns VFAT word count
+    uint8_t  getInFu() const    {return InFu_;}      ///<Returns InFIFO underflow flag
+    uint8_t  getStuckd() const  {return Stuckd_;}    ///<Returns Stuck data flag
+    std::vector<uint8_t> get_GEBflags() const {return v_GEBflags_;}
```
Comment (kpedro88) [inline DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h]: could this return a const reference?

File: DataFormats/GEMDigi/src/GEMGEBStatusDigi.cc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      OHCRC_(OHCRC),
+      Vwt_(Vwt),
+      InFu_(InFu),
+      Stuckd_(Stuckd),
+      v_GEBflags_(v_GEBflags){};
+
+
+  GEMGEBStatusDigi::GEMGEBStatusDigi(){};
+
+
```
Comment (kpedro88) [inline DataFormats/GEMDigi/src/GEMGEBStatusDigi.cc]: remove extra lines

File: DataFormats/GEMDigi/src/GEMVfatStatusDigi.cc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    b1110_(b1110),
+    lsData_(lsData),
+    msData_(msData),
+    crc_(crc),
+    crc_calc_(crc_calc),
+    isBlockGood_(isBlockGood){};
+
+  GEMVfatStatusDigi::GEMVfatStatusDigi(){};
+
+
```
Comment (kpedro88) [inline DataFormats/GEMDigi/src/GEMVfatStatusDigi.cc]: remove extra lines

File: DataFormats/GEMDigi/src/GEMGEBStatusDigi.cc [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#include "DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h"
+#include <iostream>
```
Comment (kpedro88) [inline DataFormats/GEMDigi/src/GEMGEBStatusDigi.cc]: remove unneeded include

File: DataFormats/GEMDigi/src/GEMVfatStatusDigi.cc [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#include "DataFormats/GEMDigi/interface/GEMVfatStatusDigi.h"
+#include <iostream>
```
Comment (kpedro88) [inline DataFormats/GEMDigi/src/GEMVfatStatusDigi.cc]: remove unneeded include

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
please test

File: DataFormats/GEMDigi/src/GEMVfatStatusDigi.cc (hunk 3 above)
Comment (andjuo) [reply DataFormats/GEMDigi/src/GEMVfatStatusDigi.cc]: The unneeded ';' after {} were preventing me from a successful build under CMSSW 930pre1. I am surprised no issue was reported by the builder.

**Comment by civanch** [comment]
+1

File: DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+			    uint16_t ErrorC, 
+			    uint16_t OHCRC, 
+			    uint16_t Vwt,
+			    uint8_t InFu,
+			    uint8_t Stuckd,
+			    std::vector<uint8_t> v_GEBflags);
+
+  GEMGEBStatusDigi(){}
+  
+  uint32_t getZeroSup() const {return ZeroSup_;}   ///<Returns Zero Suppression flags
```
Comment (davidlange6) [inline DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h]: hi @jshlee - please make all of these accessors const.

File: DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  uint16_t getOHCRC() const   {return OHCRC_;}     ///<Returns OH CRC 
+  uint16_t getVwt() const     {return Vwt_;}       ///<Returns VFAT word count
+  uint8_t  getInFu() const    {return InFu_;}      ///<Returns InFIFO underflow flag
+  uint8_t  getStuckd() const  {return Stuckd_;}    ///<Returns Stuck data flag
+  const std::vector<uint8_t> & get_GEBflags() const {return v_GEBflags_;}
+
+ private:
+
+  uint32_t ZeroSup_;
+  uint8_t  InputID_;
```
Comment (davidlange6) [inline DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h]: reorder in 64 bit blocks (so move InputID_ to be with the other uint8_t variables

File: DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h (hunk 6 above)
Comment (davidlange6) [reply DataFormats/GEMDigi/interface/GEMGEBStatusDigi.h]: whoops - they are const.. never mind.

File: DataFormats/GEMDigi/interface/GEMVfatStatusDigi.h [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  uint64_t getMsData() const { return msData_; } 
+  uint16_t getCrc() const { return crc_; } 
+  uint16_t getCrc_calc() const { return crc_calc_; } 
+  uint16_t getIsBlocGood() const { return isBlockGood_; } 
+
+ private:
+  uint8_t  b1010_;                   ///<1010:4 Control bits, shoud be 1010
+  uint8_t  b1100_;                   ///<1100:4, Control bits, shoud be 1100
+  uint8_t  flag_;                    ///<Control Flags: 4 bits, Hamming Error/AFULL/SEUlogic/SUEI2C
+  uint8_t  b1110_;                   ///<1110:4 Control bits, shoud be 1110
```
Comment (davidlange6) [inline DataFormats/GEMDigi/interface/GEMVfatStatusDigi.h]: similarly - group together the uint8 and uint16 variables

**Comment by kpedro88** [comment]
please test

**Comment by civanch** [comment]
please test

**Comment by mommsen** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**